pandas/tests/copy_view/test_indexing.py

```diff
@@ -462,6 +462,153 @@ def test_subset_set_with_column_indexer(
+        lambda df: df[["a", "b"]].loc[0:1],
+        lambda df: df[0:2].iloc[:, 0:2],
+        lambda df: df[0:2].loc[:, "a":"b"],
+    ],
+    ids=[
+        "row-getitem-slice",
+        "column-getitem",
+        "row-iloc-slice",
+        "row-loc-slice",
+        "column-iloc-slice",
+        "column-loc-slice",
+    ],
+)
+@pytest.mark.parametrize(
+    "dtype", ["int64", "float64"], ids=["single-block", "mixed-block"]
+)
+def test_subset_chained_getitem(
+    request, method, dtype, using_copy_on_write, using_array_manager
+):
+    # Case: creating a subset using multiple, chained getitem calls using views
+    # still needs to guarantee proper CoW behaviour
+    df = DataFrame(
+        {"a": [1, 2, 3], "b": [4, 5, 6], "c": np.array([7, 8, 9], dtype=dtype)}
+    )
+    df_orig = df.copy()
+
+    # when not using CoW, it depends on whether we have a single block or not
+    # and whether we are slicing the columns -> in that case we have a view
+    subset_is_view = request.node.callspec.id in (
+        "single-block-column-iloc-slice",
+        "single-block-column-loc-slice",
+    ) or (
+        request.node.callspec.id
+        in ("mixed-block-column-iloc-slice", "mixed-block-column-loc-slice")
+        and using_array_manager
+    )
+
+    # modify subset -> don't modify parent
+    subset = method(df)
+    subset.iloc[0, 0] = 0
+    if using_copy_on_write or (not subset_is_view):
+        tm.assert_frame_equal(df, df_orig)
+    else:
+        assert df.iloc[0, 0] == 0
+
+    # modify parent -> don't modify subset
+    subset = method(df)
+    df.iloc[0, 0] = 0
+    expected = DataFrame({"a": [1, 2], "b": [4, 5]})
+    if using_copy_on_write or not subset_is_view:
+        tm.assert_frame_equal(subset, expected)
+    else:
+        assert subset.iloc[0, 0] == 0
+
+
+@pytest.mark.parametrize(
+    "dtype", ["int64", "float64"], ids=["single-block", "mixed-block"]
+)
+def test_subset_chained_getitem_column(dtype, using_copy_on_write):
+    # Case: creating a subset using multiple, chained getitem calls using views
+    # still needs to guarantee proper CoW behaviour
+    df = DataFrame(
+        {"a": [1, 2, 3], "b": [4, 5, 6], "c": np.array([7, 8, 9], dtype=dtype)}
+    )
+    df_orig = df.copy()
+
+    # modify subset -> don't modify parent
+    subset = df[:]["a"][0:2]
+    df._clear_item_cache()
+    subset.iloc[0] = 0
+    if using_copy_on_write:
+        tm.assert_frame_equal(df, df_orig)
+    else:
+        assert df.iloc[0, 0] == 0
+
+    # modify parent -> don't modify subset
+    subset = df[:]["a"][0:2]
+    df._clear_item_cache()
+    df.iloc[0, 0] = 0
+    expected = Series([1, 2], name="a")
+    if using_copy_on_write:
+        tm.assert_series_equal(subset, expected)
+    else:
+        assert subset.iloc[0] == 0
+
+
+@pytest.mark.parametrize(
+    "method",
+    [
+        lambda s: s["a":"c"]["a":"b"],
+        lambda s: s.iloc[0:3].iloc[0:2],
+        lambda s: s.loc["a":"c"].loc["a":"b"],
```

Comment:
Could we add mixed loc and iloc?

Reply:
It _shouldn't_ be needed. In the beginning I was trying to do various combinations (for example in the first test, `test_subset_chained_getitem`, I have loc/iloc combined with a column selection and combined with a row slice). But my understanding is that what each case here is actually testing is whathever is the second indexing step (because it is this second indexing step that needs to keep the intermediate object from the first indexing step alive). So what is important is that each of the options (getitem, iloc, loc) is at least once in the second place (and in principle the two indexing steps are independent from each other, so as long as each indexing option is present both in first and second position, it should cover everything)

Now, this is of course cheap to add some extra parametrizations, but so if adding mixed loc and iloc, then I could also add all mixtures of getitem and loc/iloc, ...

Reply:
But since I was adding one extra test case with a long chain, I at the same time made this a mixed loc/iloc case!

Reply:
Thx, yeah my understanding was that this wasn't strictly necessary, but I thought cheap to add so we can cover this too.